cmd/geth/flag.go

```diff
@@ -453,6 +455,81 @@ func makeNodeName(version string, ctx *cli.Context) string {
 	return name
 }
 
+func mustMakeMLogDir(ctx *cli.Context) string {
+	if ctx.GlobalIsSet(MLogDirFlag.Name) {
+		p := ctx.GlobalString(MLogDirFlag.Name)
+		if p == "" {
+			glog.Fatalf("Flag %v requires a non-empty argument", MLogDirFlag.Name)
+			return ""
+		}
+		cwd, e := os.Getwd()
+		if e != nil {
+			cwd = "."
+		}
+		rp := common.EnsurePathAbsoluteOrRelativeTo(cwd, p)
+		if filepath.IsAbs(rp) {
+			return rp
+		}
+		ap, e := filepath.Abs(rp)
+		if e != nil {
+			glog.Fatalf("could not establish absolute path for mlog dir: %v", e)
+		}
+		return ap
+	}
+
+	rp := common.EnsurePathAbsoluteOrRelativeTo(MustMakeChainDataDir(ctx), "mlogs")
+	if !filepath.IsAbs(rp) {
```

Comment:
Again, [MustMakeChainDataDir](https://github.com/ethereumproject/go-ethereum/blob/edc500105b3da8860c6a8ea84499b7f346c9a442/cmd/geth/flag.go#L242) already calls `EnsurePathAbsoluteOrRelativeTo`. So is it possible to get a non-absolute path if the above function returns no error?

Reply:
Right -- impossible, will fix.